webui.py

```diff
@@ -618,6 +618,6 @@ def dump_default_english_config():
     server_name=args_manager.args.listen,
     server_port=args_manager.args.port,
     share=args_manager.args.share,
-    auth=check_auth if args_manager.args.share and auth_enabled else None,
+    auth=check_auth if auth_enabled else None,
```

Comment:
```suggestion
    auth=check_auth if (args_manager.args.share or args_manager.args.listen) and auth_enabled else None,
```


I don't see any reason why we should only restrict this for users using --share and not also allow basic auth protection for --listen. For this you normally would also want to rather add args_manager.args.listen than also using basic auth locally. Is there any reason why you'd also propose to have basic auth locally?

Reply:
Fair point - didn't have the local mode on screen but why not allow authentication in general and let the user decide, that's what I did here. You still actively need to create the `auth.json` file so no surprises are to be expected.